packages/graphql/src/schema/get-where-fields.ts

```diff
@@ -126,30 +124,25 @@ function getWhereFields({
             }
 
             if (["String", "ID"].includes(f.typeMeta.name)) {
-                const matchesSetting: boolean | undefined = features?.filters?.[f.typeMeta.name]?.MATCHES;
-                if (matchesSetting !== undefined) {
-                    if (matchesSetting === true) {
-                        res[`${f.fieldName}_MATCHES`] = { type: "String", directives: deprecatedDirectives };
-                    }
-                } else if (enableRegex) {
-                    // TODO: This is deprecated. To be removed in 4.0.0.
-                    res[`${f.fieldName}_MATCHES`] = { type: "String", directives: deprecatedDirectives };
-                }
-
-                const stringWhereOperators = ["_CONTAINS", "_STARTS_WITH", "_ENDS_WITH"];
+                const stringWhereOperators: { comparator: string; typeName: string }[] = [
+                    { comparator: "_CONTAINS", typeName: f.typeMeta.name },
+                    { comparator: "_STARTS_WITH", typeName: f.typeMeta.name },
+                    { comparator: "_ENDS_WITH", typeName: f.typeMeta.name },
+                ];
 
                 const stringWhereOperatorsNegate = ["_NOT_CONTAINS", "_NOT_STARTS_WITH", "_NOT_ENDS_WITH"];
 
-                Object.entries(features?.filters?.String || {}).forEach(([filter, enabled]) => {
-                    if (filter === "MATCHES") {
-                        return;
-                    }
+                Object.entries(features?.filters?.[f.typeMeta.name] || {}).forEach(([filter, enabled]) => {
```

Comment:
**nitpick** Consider creating a varaible for `f.typeMeta.name` as it is used in several places and the extra context of a explicit name could help with readability here